rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -996,6 +996,54 @@ private void logUpdateCollections(Transaction rskTx) {
         eventLogger.logUpdateCollections(sender);
     }
 
+    protected void processValidationFailure(Federation proposedFederation) {
+        eventLogger.logCommitFederationFailure(rskExecutionBlock, proposedFederation);
+        logger.warn("[processValidationFailure] Proposed federation validation failed so svp values will be cleared.");
+        clearSvpValues();
+    }
+
+    private void clearSvpValues() {
+        federationSupport.clearProposedFederation();
+
+        String methodName = "[clearSvpValues]";
+
+        Optional<Sha256Hash> svpFundTxHashUnsigned = provider.getSvpFundTxHashUnsigned();
+        if (svpFundTxHashUnsigned.isPresent()) {
+            logger.warn("{} Fund tx {} was never registered.", methodName, svpFundTxHashUnsigned.get());
+            provider.setSvpFundTxHashUnsigned(null);
+            return;
+        }
+
+        Optional<BtcTransaction> svpFundTxSigned = provider.getSvpFundTxSigned();
+        if (svpFundTxSigned.isPresent()) {
+            logger.warn("{} Spend tx was never created. Fund tx hash: {}", methodName, svpFundTxSigned.get().getHash());
+            provider.setSvpFundTxSigned(null);
+            return;
+        }
+
+        Optional<Map.Entry<Keccak256, BtcTransaction>> svpSpendTxWFSOpt = provider.getSvpSpendTxWaitingForSignatures();
+        if (svpSpendTxWFSOpt.isPresent()) {
+            Map.Entry<Keccak256, BtcTransaction> svpSpendTxWFS = svpSpendTxWFSOpt.get();
+            Keccak256 rskCreationHash = svpSpendTxWFS.getKey();
+            BtcTransaction svpSpendTx = svpSpendTxWFS.getValue();
+
+            logger.warn("{} Spend tx {} was not fully signed. Rsk creation hash: {}.",
+                methodName, svpSpendTx.getHash(), rskCreationHash);
+            provider.setSvpSpendTxWaitingForSignatures(null);
+            provider.setSvpSpendTxHashUnsigned(null);
+            return;
+        }
+
+        Optional<Sha256Hash> svpSpendTxHashUnsigned = provider.getSvpSpendTxHashUnsigned();
+        if (svpSpendTxHashUnsigned.isPresent()) {
+            logger.warn("{} Spend tx {} was not registered.", methodName, svpSpendTxHashUnsigned.get());
+            provider.setSvpSpendTxHashUnsigned(null);
+            return;
+        }
+
+        logger.error("{} All SVP values were already clear, so validation should have been successful.", methodName);
```

Comment:
I think it could be simplier if we just attempt to clear everything one by one and we can avoid this log statement and have an INFO one at the end saying something like "all SVP value have been wiped etc"

Reply:
same as [here](https://github.com/rsksmart/rskj/pull/2817#discussion_r1810763724)